### PR TITLE
support prefix tuning for starcoder models

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Support Prefix Tuning for StarCoder models.

### CODE REVIEW

File: src/peft/utils/other.py [hunk 1]
```diff
@@ -460,6 +471,7 @@ def get_auto_gptq_quant_linear(gptq_quantization_config):
 
 TRANSFORMERS_MODELS_TO_PREFIX_TUNING_POSTPROCESS_MAPPING = {
     "bloom": bloom_model_postprocess_past_key_value,
+    "gpt_bigcode": starcoder_model_postprocess_past_key_value,
```
Comment (BenjaminBossan) [inline src/peft/utils/other.py:474]: Can we be sure that this model type will only match starcoder and not, for instance, santacoder? Or is the post-processing the same?

**Review by BenjaminBossan — APPROVED** [review]
Thanks for adding this capability. I assume that you have tested it and it works. Adding a unit test is probably not trivial here, so I'm okay as is. There is only one small question I added.

File: src/peft/utils/other.py (hunk 1 above)
Comment (pacman100) [reply src/peft/utils/other.py:474]: Hello, `santacoder` is just a smaller version, i.e.,  post-processing is the same

**Comment by pacman100** [comment]
> I assume that you have tested it and it works. Adding a unit test is probably not trivial here

Should be addressed in the latest commit. Also, I've tested this.

**Comment by BenjaminBossan** [comment]
Ah I see, thanks for explaining. Also very nice that the test coverage is now increased.

**Pull request merged**